src/amo/components/Addon/index.js

```diff
@@ -452,8 +453,14 @@ export class AddonBase extends React.Component {
       isRestartRequired = addon.isRestartRequired;
     }
 
+    const addonsByAuthorsLength = addonsByAuthors ? addonsByAuthors.length : 0;
```

Comment:
Would `numberOfAddonsByAuthor` be a bit more descriptive? Otherwise just make `addonsByAuthors` always an array (with a default value or something?) so you can always do `.length` on it because it's a bit weird to see what is basically `addonsByAuthorsLength = addonsByAuthors.length` 😄